.github/Dockerfile.debian11-test

```diff
@@ -0,0 +1,3 @@
+FROM public.ecr.aws/debian/debian:11
+
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get -y install --no-install-recommends build-essential debhelper git devscripts shellcheck
```

Comment:
The .deb build process runs the test suite, which in this case involves running shellcheck, which is why it's there (and listed as a build-dependency in [debian/control](https://github.com/nmeyerhans/amazon-ec2-net-utils/commit/0bc8a67089fab65d1c5c401ef1e9689d48925f5c)). I'd like to run the tests during rpm build as well, but shellcheck isn't in AL2022 yet.

Because of this, we run shellcheck twice, but having it as a discrete step is useful for visibility in the workflow runs.